Yoast/yoast-acf-analysis
# Remove ACF4 code

## Summary

This PR can be summarized in the following changelog entry:

* Remove all ACF 4-specific code.

## Relevant technical choices:

* the JS test is not actually working. The ACF4 tests have been removed from there anyway.

## Test instructions

This PR can be tested by following these steps:

* Install ACF4, the latest release for the 4.* branch can be downloaded [here](https://downloads.wordpress.org/plugin/advanced-custom-fields.4.4.12.zip)
* Install Yoast SEO
* checkout this branch and activate
* create a new group of custom fields for posts in ACF, sticking to just the ones that allow you to add texts (they should be Textbox, Textarea, Email, WYSIWYG)
* edit a post, and see that adding text in those fields doesn't change the number of words counted by the SEO analysis
* see alsto that everything else is working and there are no fatal errors in the console that break the behaviour of the block editor, or of Yoast SEO.
* perform the same test with Classic Editor.
* delete ACF4 and installa ACF5 Free (you can do it from Plugins→Add new, otherwise download [from here](https://downloads.wordpress.org/plugin/advanced-custom-fields.5.9.1.zip))
* create a new group of custom fields for posts in ACF, you can import the ZIP that you can find in [this PR](https://github.com/Yoast/yoast-acf-analysis/pull/275), rememeber that Repeater and Flexible types won't be working.
* test that the analysis is working (follow the suggestions to test #275) and there are no console errors.
* perform the same test with Classic Editor.
* Remove ACF5 and install ACF 5 Pro, this would enable Repeater and Flexible fields if you are using the zip from above
* test that the analysis is working (follow the suggestions to test #275) and there are no console errors.
* perform the same test with Classic Editor.


Fixes [P3-167]
